lib/sql.ts

```diff
@@ -0,0 +1,208 @@
+/**
+ * Async worker thread wrapper around SQLite, written to improve concurrent performance.
+ * @author mia-pi-git
+ */
+import {ProcessWrapper, ProcessManager} from './process-manager';
+import * as Sqlite from 'better-sqlite3';
+import * as child_process from 'child_process';
+import * as path from 'path';
+import {FS} from './fs';
+
+interface SQLOptions {
+	file: string;
+	/** file to import database functions from - this should be relative to this filename. */
+	extension?: string;
+}
+
+type DataType = unknown[] | Record<string, unknown>;
+
+export type DatabaseQuery = {
```

Comment:
This seems like an implementation detail to me, do we really need to export it?

Reply:
Probably not, but I can't see it hurting.

Reply:
I don't like it when implementation details are exported; if there's anything that would change if you switched from `better-sqlite3` to `sqlite`, then don't let any consumers use it. It's a code smell sort of thing, your API is messed up if anyone needs to know what this type is

Reply:
Alright.